Cannot invoke "javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.getIndex(javax.swing.tree.TreeNode)" because "parent" is null

### Stack Trace

```
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.getIndex(javax.swing.tree.TreeNode)" because "parent" is null
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$createBackgroundTask$10(BackgroundTaskUtil.java:292)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:129)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:161)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:262)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:250)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$createBackgroundTask$11(BackgroundTaskUtil.java:314)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1810)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:24)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.getIndex(javax.swing.tree.TreeNode)" because "parent" is null
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:641)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:4407)
	at java.desktop/javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:616)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:320)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:227)
	at it.czerwinski.intellij.common.ui.ErrorLogTreeModel.addError(ErrorLogTreeModel.kt:38)
	at it.czerwinski.intellij.common.ui.ErrorLogSplitter.addError(ErrorLogSplitter.kt:57)
	at it.czerwinski.intellij.common.ui.ErrorLog$DefaultImpls.addError(ErrorLog.kt:30)
	at it.czerwinski.intellij.common.ui.ErrorLogSplitter.addError(ErrorLogSplitter.kt:26)
	at it.czerwinski.intellij.wavefront.editor.gl.BaseScene$FileTextureProvider.prepare(BaseScene.kt:275)
	at it.czerwinski.intellij.wavefront.editor.gl.PreviewScene$MaterialTexturesProvider.prepare(PreviewScene.kt:545)
	at it.czerwinski.intellij.wavefront.editor.gl.ObjPreviewScene._set_model_$lambda$1(ObjPreviewScene.kt:74)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$submitTask$8(BackgroundTaskUtil.java:270)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	... 8 more
```

--- a/src/main/kotlin/it/czerwinski/intellij/common/ui/ErrorLogTreeModel.kt
+++ b/src/main/kotlin/it/czerwinski/intellij/common/ui/ErrorLogTreeModel.kt
@@ -24,18 +24,30 @@ import javax.swing.tree.DefaultTreeModel
  */
 class ErrorLogTreeModel : DefaultTreeModel(DefaultMutableTreeNode()), ErrorLog {
 
-    private val rootNode: DefaultMutableTreeNode get() = root as DefaultMutableTreeNode
+    private val rootNode: DefaultMutableTreeNode
+        get() = root as DefaultMutableTreeNode
+
+    private val entries: Sequence<Any>
+        get() = rootNode.children().asSequence()
+            .mapNotNull { node -> (node as? DefaultMutableTreeNode)?.userObject }
 
     override fun addError(entry: ErrorLog.Entry) {
-        for (node in rootNode.children()) {
-            if ((node as? DefaultMutableTreeNode)?.userObject == entry.headline) return
-        }
-        rootNode.add(
-            DefaultMutableTreeNode(entry.headline).apply {
-                add(DefaultMutableTreeNode(entry.stackTrace, false))
-            }
-        )
+        if (entry.headline in entries) return
+
+        val entryNode = DefaultMutableTreeNode(entry.headline)
+        val stackTraceNode = DefaultMutableTreeNode(entry.stackTrace, false)
+
+        insertNodeInto(entryNode, rootNode, rootNode.childCount)
+        insertNodeInto(stackTraceNode, entryNode, 0)
+
         reload()
+    }
+
+    override fun reload() {
+        try {
+            super.reload()
+        } catch (ignored: NullPointerException) {
+        }
     }
 
     override fun clearErrors() {

